### PR TITLE
[hotfix] [connector-cdc-oracle ] support read partition table

### LINKED ISSUE #8229: [Bug] [connector-cdc-oracle ] Unsupport read partition table
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

oracle实时监听不到分区表,历史数据也不会同步，实时数据也不会监听
image
在flinkcdc中已经修复该问题
image

### SeaTunnel Version

2.3.7，2.3.8

### SeaTunnel Config

```conf
env {
  job.mode = "STREAMING"
  parallelism = 1
  checkpoint.interval = 10000
  read_limit.rows_per_second=40960
}
source {
 Oracle-CDC {
    base-url = "jdbc:oracle:thin:@//10.9.3.71:1521/DEV"
    username = "logminer_erp"
    password = "NxJ89TX0Bbc"
    database-names = ["DEV"]
    schema-names = ["XLA"]
    table-names = ["DEV.XLA.XLA_AE_LINES"]
    table-names-config = [
      {
        table = "DEV.XLA.XLA_AE_LINES",
        primaryKeys = ["APPLICATION_ID","AE_HEADER_ID","AE_LINE_NUM"]
      }

    ]

    source.reader.close.timeout = 120000
    skip_analyze = true
    use_select_count = true
    exactly_once = false
  }
}


sink {
  Doris {
    fenodes = "zxxx"
    username = root
    password = "888888g"
    database = "oracle_cdc"
    table = "${table_name}"
    sink.label-prefix = "prod-oracle"
    sink.enable-2pc = "false"
    sink.enable-delete = "true"
    generate_sink_sql = true
    doris.batch.size = 40960
    doris.config {
      format = "json"
      read_json_by_line = "true"
    }
    schema_save_mode = "CREATE_SCHEMA_WHEN_NOT_EXIST"
    data_save_mode = "APPEND_DATA"
    save_mode_create_template = """
      CREATE TABLE IF NOT EXISTS `${database}`.`${table_name}` (
      ${rowtype_primary_key},
      ${rowtype_fields}
      ) ENGINE=OLAP
        UNIQUE KEY (${rowtype_primary_key})
        DISTRIBUTED BY HASH (APPLICATION_ID)
        PROPERTIES (
          "replication_allocation" = "tag.location.default: 1",
          "in_memory" = "false",
          "storage_format" = "V2",
          "disable_auto_compaction" = "false"
        )
      """
  }
}
```

### Running Command

```shell
集群模式启动
```

### Error Exception

```log
无错误
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangshenghang**
### Purpose of this pull request

- close #8229 

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
waiting for this pr start ci
https://github.com/apache/seatunnel/pull/8262

**Comment by dailai** [comment]
Please use rebase to merge the code

**Comment by zhangshenghang** [comment]
@hailin0 @dailai @Carl-Zhou-CN @Hisoka-X 

I encountered an issue where the update data in the e2e is:

﻿> Insert delete can run successfully

image
﻿
After debugging and printing the logs, I found that Debezium modified it to the following:
﻿
`update "DEBEZIUM"."PARTITION_SOURCE_TABLE" set "VAL_VARCHAR" = 'vc3' where "VAL_VARCHAR" = 'vc2';`
﻿
The full content:
﻿
```bash
LogMinerEventRow{scn=2168886, tableId='ORCLCDB.DEBEZIUM.PARTITION_SOURCE_TABLE', tableName='PARTITION_SOURCE_TABLE', tablespaceName='DEBEZIUM', eventType=UPDATE, changeTime=2024-12-11T10:53:49Z, transactionId='0a000700fe040000', operation='UPDATE', userName='DEBEZIUM', rowId='AAAR4PAAHAAABGeAAA', rollbackFlag=false, rsId= 0x000006.0000264f.0010 , ssn=0, redoSql='update "DEBEZIUM"."PARTITION_SOURCE_TABLE" set "VAL_VARCHAR" = 'vc3' where "VAL_VARCHAR" = 'vc2';'}
```
﻿

However, for non-partitioned tables, Debezium can correctly capture the SQL.
﻿
image
﻿
I wonder if this is a bug in Debezium?
﻿
Flink CDC also does not support this feature, although there is PR, it has not been merged: https://github.com/apache/flink-cdc/pull/2479



Looking forward to your advice！

**Comment by hailin0** [comment]
Try to start logminer from sql command line to capture sql logs?

ref
https://www.modb.pro/db/394933

**Comment by hailin0** [comment]
Try to start logminer query sql logs form sql cmd

ref
https://www.modb.pro/db/394933

**Comment by hailin0** [comment]
I tested everything is ok

image


```java
2024-12-25 18:42:54,602 INFO  org.apache.seatunnel.connectors.seatunnel.console.sink.ConsoleSinkWriter - subtaskIndex=0  rowIndex=4:  SeaTunnelRow#tableId=ORCL.QA_SOURCE.TEST_PARTITION SeaTunnelRow#kind=UPDATE_BEFORE : 1, 1, true
2024-12-25 18:42:54,602 INFO  org.apache.seatunnel.connectors.seatunnel.console.sink.ConsoleSinkWriter - subtaskIndex=0  rowIndex=5:  SeaTunnelRow#tableId=ORCL.QA_SOURCE.TEST_PARTITION SeaTunnelRow#kind=UPDATE_AFTER : 1, 1, false
2024-12-25 18:43:44,499 INFO  org.apache.seatunnel.connectors.seatunnel.console.sink.ConsoleSinkWriter - subtaskIndex=0  rowIndex=6:  SeaTunnelRow#tableId=ORCL.QA_SOURCE.TEST_PARTITION SeaTunnelRow#kind=UPDATE_BEFORE : 2, 1, true
2024-12-25 18:43:44,499 INFO  org.apache.seatunnel.connectors.seatunnel.console.sink.ConsoleSinkWriter - subtaskIndex=0  rowIndex=7:  SeaTunnelRow#tableId=ORCL.QA_SOURCE.TEST_PARTITION SeaTunnelRow#kind=UPDATE_AFTER : 2, 1, false
2024-12-25 18:51:30,481 INFO  org.apache.seatunnel.connectors.seatunnel.console.sink.ConsoleSinkWriter - subtaskIndex=0  rowIndex=8:  SeaTunnelRow#tableId=ORCL.QA_SOURCE.TEST_PARTITION SeaTunnelRow#kind=DELETE : 1, 1, false

```

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-oracle/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/utils/OracleConnectionUtils.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public static List<TableId> listTables(
         Set<TableId> tableIdSet = new HashSet<>();
         String queryTablesSql =
                 "SELECT OWNER ,TABLE_NAME,TABLESPACE_NAME FROM ALL_TABLES \n"
-                        + "WHERE TABLESPACE_NAME IS NOT NULL AND TABLESPACE_NAME NOT IN ('SYSAUX')";
+                        + "WHERE PARTITIONED = 'YES' OR TABLESPACE_NAME IS NOT NULL AND TABLESPACE_NAME NOT IN ('SYSAUX')";
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-oracle/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/oracle/utils/OracleConnectionUtils.java]:
```suggestion
                        + "WHERE PARTITIONED = 'YES' OR (TABLESPACE_NAME IS NOT NULL AND TABLESPACE_NAME NOT IN ('SYSAUX'))";
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-oracle-e2e/src/test/resources/oraclecdc_to_oracle_with_partition.conf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    password = "top_secret"
+    database-names = ["ORCLCDB"]
+    schema-names = ["DEBEZIUM"]
+    table-names = ["ORCLCDB.DEBEZIUM.PARTITION_SOURCE_TABLE"]
+    base-url = "jdbc:oracle:thin:@oracle-host:1521/ORCLCDB"
+    source.reader.close.timeout = 120000
+    connection.pool.size = 1
+    debezium {
+       # log.mining.strategy = "online_catalog"
+       # log.mining.continuous.mine = true
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-oracle-e2e/src/test/resources/oraclecdc_to_oracle_with_partition.conf]:
```suggestion
```

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
LGTM if ci passes

**Pull request merged**